src/fetcher/create-breadcrumbs.tsx

```diff
@@ -15,6 +15,17 @@ type TaxonomyTermNodes = Extract<
   { __typename: 'Article' }
 >['taxonomyTerms']['nodes']
 
+function countParents(taxonomyPath: TaxonomyTermNodes[0]) {
+  function count(current: RecursiveTree): number {
+    if (current.parent) {
+      return count(current.parent) + 1
+    } else {
+      return 0
+    }
+  }
+  return count(taxonomyPath as RecursiveTree)
+}
+
```

Comment:
there are some refactorings that come to mind (I think this could be a good use case for `reduce`.
But at least `return current.parent ? count(current.parent) + 1 : 0` would look nicer ;)

Reply:
This is everyday work, not a code golf competition ;) this is code I would teach in high school

Besides of the textbook-like look: is there anything actually unclear what this function is doing that would require a refactoring?